internal/codegen/template.go

```diff
@@ -64,31 +64,51 @@ type Template struct {
 	// Contents is the content of this template
 	Contents []byte
 
+	// Binary denotes if a template is a binary "template" or not.  Binary
+	// templates only generate the single binary file for output.
```

Comment:
**thought(non-blocking):** Might be worth denoting that binary in this case means it is not processed at all.